includes/admin/settings/class.llms.settings.page.php

```diff
@@ -14,40 +14,70 @@
  * @since 1.0.0
  * @since 3.30.3 Explicitly define class properties.
  * @since 3.35.0 Unslash input data.
+ * @since [version] Add a constructor which registers the settings page and automatically saves and outputs settings content.
+ *               Add public method stub `get_label()` which is used to automatically set the `$label` property on class initialization.
+ *               Add utility method to generate a group of settings.
  */
 class LLMS_Settings_Page {
 
 	/**
 	 * Allow settings page to determine if a rewrite flush is required
 	 *
-	 * @var      boolean
-	 * @since    3.0.4
-	 * @version  3.0.4
+	 * @var boolean
 	 */
 	protected $flush = false;
 
 	/**
 	 * Settings identifier
 	 *
-	 * @var      string
-	 * @since    1.0.0
+	 * @var string
 	 */
-	public $id;
+	public $id = '';
 
 	/**
-	 * Settings page link label
+	 * Settings page label / title.
 	 *
-	 * @var      string
-	 * @since    1.0.0
+	 * @var string
 	 */
-	public $label;
+	public $label = '';
```

Comment:
do we need to set this to `''`?
My understanding is that the first thing we do is to set the `$label` property in the constructor so...

And now let me be a pain in the... I don't like the fact that the method `get_label()` doesn't actually return the `$label` property value, it's confusing...
I know that you created that method because you need to run the gettext function plus you wanted to make the override as small as possible, but let's use a different name, not sure... `build_label()` ? Maybe `protected` too?

What do you think? I know this is not the most important thing, but still I think it's better to change it now, if you agree.

Reply:
I don't think we have to set it but I like the idea of being able to say *this is literally always a string* which we can't if we don't set it. There is a world in which someone using the API decides they're not going to set it and I guess it's their problem then but I still think setting it as an empty string makes logical sense.

You're right about `get_label()` being wrong. I'm going to change it to `set_label()`